ovirt_imageio/client/_upload.py

```diff
@@ -0,0 +1,156 @@
+# ovirt-imageio
+# Copyright (C) 2022 Red Hat, Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+"""
+Upload commands.
+"""
+
+import os
+from contextlib import closing
+from collections import namedtuple
+
+from . import _api
+from . import _options
+from . import _ovirt
+from . import _ui
+
+
+DiskInfo = namedtuple(
+    "DiskInfo",
+    "name,initial_size,provisioned_size,content_type,format,sparse")
+
+
+def register(parser):
+    cmd = parser.add_sub_command(
+        "upload-disk",
+        help="Upload disk",
+        func=upload_disk)
+
+    cmd.add_argument(
+        "-s", "--storage-domain",
+        required=True,
+        help="Name of the storage domain.")
+
+    cmd.add_argument(
+        "-f", "--format",
+        choices=("raw", "qcow2"),
+        default="qcow2",
+        help="Upload disk format (default qcow2 for data disks and raw "
+             "for iso disks).")
+
+    cmd.add_argument(
+        "--preallocated",
+        dest="sparse",
+        action="store_false",
+        help="Create preallocated disk. Required when using raw format on "
+             "block based storage domain (iSCSI, FC). ISO images are "
+             "always uploaded to preallocated disk.")
+
+    cmd.add_argument(
+        "--id", "--disk-id",
```

Comment:
Adding 2 long arguments is not common. Lets use only --disk-id which is consistent with disk_id (for download) and is more clear.

This will also make it more clear what is the argument name (args.id or args.disk_id)?
Is this even documented behavior?

Reply:
`Is this even documented behavior?` Yes, it takes the dest name from the first long option, stripping away the initial `--`, so in this case it would be `id`.

https://docs.python.org/3/library/argparse.html#dest

Anyway, I'm ok with leaving it as `--disk-id`, specially as it is consistent with download command (sortof at least, since for download it is positional argument, but still). I'll remove `id` for the next version.